typescript/sdk/src/hook/HyperlaneHookDeployer.ts

```diff
@@ -1,169 +1,100 @@
 import debug from 'debug';
 
-import {
-  OptimismISM,
-  OptimismISM__factory,
-  OptimismMessageHook,
-  OptimismMessageHook__factory,
-  TestRecipient,
-  TestRecipient__factory,
-} from '@hyperlane-xyz/core';
 import { types } from '@hyperlane-xyz/utils';
-import { addressToBytes32 } from '@hyperlane-xyz/utils/dist/src/utils';
 
 import { HyperlaneContracts, HyperlaneContractsMap } from '../contracts';
 import { HyperlaneDeployer } from '../deploy/HyperlaneDeployer';
 import { MultiProvider } from '../providers/MultiProvider';
 import { ChainMap, ChainName } from '../types';
+import { objFilter, objMap, promiseObjAll } from '../utils/objects';
 
 import { isHookConfig, isISMConfig } from './config';
-import { HookFactories, hookFactories } from './contracts';
-import { HookConfig } from './types';
+import { OptimismHookFactories, optimismHookFactories } from './contracts';
+import { HookConfig, MessageHookConfig, NoMetadataIsmConfig } from './types';
 
+// TODO: make generic from optimism hooks
```

Comment:
Seems big enough we may want a ticket to track this

Reply:
@aroralanuk is on it